HIP/hip-551.md

```diff
@@ -2,126 +2,210 @@
+1. Each inner transaction within a batch should be treated as a self-contained transaction.
+2. If an inner transaction fails, preceding transactions that succeeded will still incur fees, even though their effects are not committed.
 3. Inner transactions are individually signed regular transactions indicating that once a person signs an inner transaction, they have agreed to its outcome.
 
-
 ### Batch Signatures
 
-Each transaction must be individually signed. Signatures on the batch itself are unique to the batch and don't represent the inner transactions within. This way:
+- Batch transactions require unique signatures for the batch itself. Inner transactions cannot be rearranged post-signature.
+- When someone signs a transaction, they individually agree to the outcome, thus the individual transaction signers within the batch need not sign the batch itself.
+
+### BatchKey Introduction
 
-- It's impossible for inner transactions to be rearranged post-signature.
-- Batch transactions are not designed to solve a problem for retail users but rather for developers and dApps. When someone signs a transaction, they individually agree to the outcome, thus the individual transaction signers within the batch need not sign the batch itself.
-  
-### Handling Inner Transactions
+- To prevent malicious actors from tampering with a batch (e.g., removing, reordering, or adding transactions), a  `batchKey` will be introduced in the `TransactionBody`.
+- The `batchKey` signals the trusted signer who can finalize the batch. This key ensures that batch transactions can only be submitted as a whole and prevents individual transactions from being executed outside the batch context.
+- `batchKey` is required for all inner transactions. Any batch which contains an inner transaction without a `batchKey` set to a valid public key will be rejected.
+- The top-level `AtomicBatchTransactionBody` must be signed by all `batchKey` keys specified in the included transactions. i.e. if inner transaction X sets `batchKey` X and inner transaction Y sets `batchKey` Y then their containing AtomicBatch transaction must be signed by both `batchKey` X and `batchKey` Y private keys.
+- `batchKey` is not allowed to be set on a transaction that is not an inner transaction. Any top-level transaction that has a value set for `batchKey` will be rejected.
+
+### Inner vs Outer Transactions
 
 Inner transactions, by design:
 
 1. Are signed individually.
 2. Each pays for itself (node + network + service fees).
 3. Have their own payer.
-4. Are deduplicated on an individual basis.
-5. Are authorized solely by their keys; batch keys are not used.
-6. If one of the inner transactions has a child transaction, the parent transaction is in the batch, and the batch isn't successful, then the child transaction should also be reverted.
+4. Are deduplicated on an individual basis. If two inner transactions within the same batch have the same transactionID, the transaction will be rejected.
```

Comment:
A TransactionID has many components. To be clear, they each should provide a unique combination of accountID plus transactionValidStart? They should still continue to set scheduled=false and nonce=0, right?

Reply:
Maybe we can add a short comment to the HIP to clarify that scheduled=false and nonce=0 for all inner transactions.